yaml/secret.go

```diff
@@ -36,7 +36,7 @@ type (
 		Name        string                 `yaml:"name,omitempty"        json:"name,omitempty" jsonschema:"required,minLength=1,description=Unique name for the secret origin."`
 		Parameters  map[string]interface{} `yaml:"parameters,omitempty"  json:"parameters,omitempty" jsonschema:"description=Extra configuration variables for the secret plugin.\nReference: coming soon"`
 		Secrets     StepSecretSlice        `yaml:"secrets,omitempty"     json:"secrets,omitempty" jsonschema:"description=Secrets to inject that are necessary to retrieve the secrets.\nReference: coming soon"`
-		Pull        bool                   `yaml:"pull,omitempty"        json:"pull,omitempty" jsonschema:"description=Automatically upgrade to the latest version of the image.\nReference: coming soon"`
+		Pull        string                 `yaml:"pull,omitempty"        json:"pull,omitempty" jsonschema:"description=Automatically upgrade to the latest version of the image.\nReference: coming soon"`
```

Comment:
Converting the `Pull` field to a `string` for the `github.com/go-vela/types/yaml.Origin` type

Reply:
Should probably update the `jsonschema` to reflect the newly available options.

Reply:
Did you mean to update the secret stuff? The description only mentions image pull policy.

Reply:
@JordanSussman yeah I kept it specifically to referencing an `image` pull policy and not the Vela resource types because this change is being made in multiple places. For example, now this is used in 3 distinct types:

* `secrets`
* `services`
* `steps`

If I were to update the PR description, I think I'd have to mention all 3, not just `services` and `secrets` 👍 

I can see the original issue is explicitly calling out steps so I'll update that issue.